packages/assets-controllers/src/NftController.ts

```diff
@@ -512,6 +512,53 @@ export class NftController extends BaseController<
     });
   }
 
+  /**
+   * Fetches NFT Collection Metadata from the NFT API.
+   *
+   * @param contractAddresses - The contract addresses of the NFTs.
+   * @param chainId - The chain ID of the network where the NFT is located.
+   * @returns NFT collections metadata.
+   */
+  async fetchNftCollectionsMetadata(
+    contractAddresses: string[],
+    chainId: Hex,
+  ): Promise<{
+    collections: Collection[];
+  }> {
+    return await this.#getNftCollectionInformationFromApi(
+      contractAddresses,
+      chainId,
+    );
+  }
+
+  async #getNftCollectionInformationFromApi(
+    contractAddress: string[],
+    chainId: Hex,
+  ) {
+    const urlParams = new URLSearchParams({
+      chainId,
+    });
+
+    for (const address of contractAddress) {
+      urlParams.append('contract', address);
+    }
+
+    return await handleFetch(
```

Comment:
Beyond the scope of this PR, but I believe an upcoming decision is to encourage the encapsulation of APIs in their own service classes.

Perhaps a future task to migrate all of the various NFT APIs to their own classes, but still utilised and exposed by this controller?

Reply:
That would definitely make the controller look cleaner and easy to read.